cedar-policy-core/src/evaluator.rs

```diff
@@ -1346,6 +1451,216 @@ pub mod test {
         );
     }
 
+    #[test]
+    fn interpret_entity_tags() {
+        let request = basic_request();
+        let entities = rich_entities();
+        let eval = Evaluator::new(request, &entities, Extensions::none());
+        // hasTag on an entity with no tags
+        assert_eq!(
+            eval.interpret_inline_policy(&Expr::has_tag(
+                Expr::val(EntityUID::with_eid("entity_no_attrs_no_parents")),
+                Expr::val("doesnotexist"),
+            )),
+            Ok(Value::from(false))
+        );
+        // hasTag on an entity that has tags, but not that one
+        assert_eq!(
+            eval.interpret_inline_policy(&Expr::has_tag(
+                Expr::val(EntityUID::with_eid("entity_with_tags")),
+                Expr::val("doesnotexist"),
+            )),
+            Ok(Value::from(false))
+        );
+        // hasTag where the response is true
+        assert_eq!(
+            eval.interpret_inline_policy(&Expr::has_tag(
+                Expr::val(EntityUID::with_eid("entity_with_tags")),
+                Expr::val("spoon"),
+            )),
+            Ok(Value::from(true))
+        );
+        // hasTag, with a computed key, where the response is true
+        assert_eq!(
+            eval.interpret_inline_policy(&Expr::has_tag(
+                Expr::val(EntityUID::with_eid("entity_with_tags")),
+                Expr::get_attr(
+                    Expr::val(EntityUID::with_eid("entity_with_tags_and_attrs")),
+                    "fork".into()
+                ),
+            )),
+            Ok(Value::from(true))
+        );
+        // getTag on a tag which doesn't exist (and no attrs exist)
+        assert_matches!(
+            eval.interpret_inline_policy(&Expr::get_tag(
+                Expr::val(EntityUID::with_eid("entity_with_tags")),
+                Expr::val("doesnotexist"),
+            )),
+            Err(EvaluationError::EntityAttrDoesNotExist(e)) => {
+                let report = miette::Report::new(e.clone());
+                assert_eq!(e.entity.as_ref(), &EntityUID::with_eid("entity_with_tags"));
+                assert_eq!(&e.attr_or_tag, "doesnotexist");
+                let available_attrs = e.available_attrs_or_tags;
+                assert_eq!(available_attrs.len(), 1);
+                assert!(available_attrs.contains(&"spoon".into()));
+                expect_err(
+                    "",
+                    &report,
+                    &ExpectedErrorMessageBuilder::error(r#"`test_entity_type::"entity_with_tags"` does not have the tag `doesnotexist`"#)
+                        .help("available tags: [spoon]")
+                        .build()
+                );
+            }
+        );
+        // getTag on a tag which doesn't exist (but the corresponding attr does)
```

Comment:
I don't think I see a corresponding test case for `hasTag`

Reply:
it's at line 1467, I suppose, since hasTag doesn't care whether the corresponding attr exists or not.  But happy to add a specific analogue to this for hasTag.

Reply:
Added another test